Cabal/Distribution/Version.hs

```diff
@@ -115,6 +113,93 @@ import Text.PrettyPrint ((<+>))
 import Control.Exception (assert)
 
 -- -----------------------------------------------------------------------------
+-- Versions
+
+-- | A 'Version' represents the version of a software entity.
+--
+-- Instances of 'Eq' and 'Ord' are provided, which gives exact
+-- equality and lexicographic ordering of the version number
+-- components (i.e. 2.1 > 2.0, 1.2.3 > 1.2.2, etc.).
+--
+-- This type is opaque and distinct from 'Base.Version' in @base@
+-- since @Cabal-2.0@.
+--
+-- @since 2.0
+data Version = Version [Int]
+             deriving (Data,Eq,Ord,Generic,Show,Read,Typeable)
+
+instance Binary Version
```

Comment:
Should we say it explicitly we aren't going to preserve `Binary` compatibility (pun unintended)